utils/cloudinfo/aws.go

```diff
@@ -23,7 +23,10 @@ import (
 )
 
 func onAWS() bool {
-	client := ec2metadata.New(session.New(&aws.Config{}))
+	// the default client behavior retried the operation 4 times with a 5s timeout per attempt.
+	// if you were not on aws, you would block for 20s when invoking this operation.
+	// we reduce retries to 0 to reduce the time this blocks when not on aws.
+	client := ec2metadata.New(session.New(&aws.Config{MaxRetries: aws.Int(0)}))
```

Comment:
yes, but I was apprehensive to doing that since I have no idea if the 5s was based on actual observation in aws.  that said, i would be happy to go to 2s or 1s if someone could vouch for that risk.

`client.Config.HTTPClient.Timeout` can be lowered after this line to something smaller...